pandas/tests/io/json/test_pandas.py

```diff
@@ -470,101 +471,99 @@ def _check_all_orients(
         # force everything to have object dtype beforehand
         _check_orient(df.transpose().transpose(), "index", dtype=False)
 
-    def test_frame_from_json_bad_data(self):
-        with pytest.raises(ValueError, match="Expected object or value"):
-            read_json(StringIO('{"key":b:a:d}'))
-
-        # too few indices
-        json = StringIO(
-            '{"columns":["A","B"],'
-            '"index":["2","3"],'
-            '"data":[[1.0,"1"],[2.0,"2"],[null,"3"]]}'
-        )
-        msg = r"Shape of passed values is \(3, 2\), indices imply \(2, 2\)"
+    @pytest.mark.parametrize(
+        "data,msg,orient",
+        [
+            (StringIO('{"key":b:a:d}'), "Expected object or value", "columns"),
+            # too few indices
+            (
+                StringIO(
+                    '{"columns":["A","B"],'
+                    '"index":["2","3"],'
+                    '"data":[[1.0,"1"],[2.0,"2"],[null,"3"]]}'
+                ),
+                r"Shape of passed values is \(3, 2\), indices imply \(2, 2\)",
+                "split",
+            ),
+            # too many columns
+            (
+                StringIO(
+                    '{"columns":["A","B","C"],'
+                    '"index":["1","2","3"],'
+                    '"data":[[1.0,"1"],[2.0,"2"],[null,"3"]]}'
+                ),
+                "3 columns passed, passed data had 2 columns",
+                "split",
+            ),
+            # bad key
+            (
+                StringIO(
```

Comment:
can/should the StringIO calls go inside the test?

Reply:
Yea I think that would make it more readable - nice idea!